rpc/src/rpc.rs

```diff
@@ -610,12 +610,20 @@ impl JsonRpcRequestProcessor {
         // epoch
         let bank = self.get_bank_with_config(context_config)?;
 
-        // DO NOT CLEAN UP with feature_set::enable_partitioned_epoch_reward
+        // DO NOT CLEAN UP with feature_set::partitioned_epoch_rewards_superfeature
         // This logic needs to be retained indefinitely to support historical
         // rewards before and after feature activation.
         let partitioned_epoch_reward_enabled_slot = bank
```

Comment:
Can't we look at the first block and check if there are stake rewards there to see whether partitioned rewards was enabled?

Reply:
That sounds possible, though I'll have to think about how to do that performatively, since it flops the existing logic ing `filter_reward_maps`...

Actually, a better approach might be to read `num_reward_partitions` from the epoch-boundary block to determine if partitioned rewards was enabled. That field is recorded as `Some` for the epochs currently elapsing on testnet, but should go back to being populated as `None` after deactivation.

Any objection to me merging this PR as-is and then following up with such an RPC improvement?

Reply:
No objection, approval still stands